Diary/Controller/DiaryDetailViewController.swift

```diff
@@ -7,59 +7,159 @@
         return textView
     }()
-
+    
     private let bodyTextView: UITextView = {
         let textView = UITextView()
         textView.font = .preferredFont(forTextStyle: .subheadline)
         textView.translatesAutoresizingMaskIntoConstraints = false
         return textView
     }()
-
-    init(diaryItem: Diary) {
-        self.diaryItem = diaryItem
+    
+    init(diaryData: Diary? = nil) {
+        self.diaryData = diaryData
         super.init(nibName: nil, bundle: nil)
     }
-
+    
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self, name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        setupNavigationBar()
-        configureCellData()
+        configureNavigationBar()
         setupLayout()
+        setData()
+        setFirstResponder()
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide), name: UIResponder.keyboardWillHideNotification, object: nil)
     }
-
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        guard !isDeleted else { return }
+        
+        saveOrUpdateDiary()
+    }
+    
     override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
-        self.view.endEditing(true)
+        view.endEditing(true)
     }
-
-    private func setupNavigationBar() {
-        title = DateToStringFormatter.changeToString(from: diaryItem.createdAt)
+    
+    private func configureNavigationBar() {
+        navigationItem.rightBarButtonItem =
+        UIBarButtonItem(image: UIImage(systemName: "ellipsis.circle"), style: .plain,
+                        target: self, action: #selector(tappedDetailButton))
     }
-
-    private func configureCellData() {
-        titleTextView.text = diaryItem.title
-        bodyTextView.text = diaryItem.body
+    
+    @objc func keyboardWillHide(notification: Notification) {
+        print(#function)
+        saveOrUpdateDiary()
     }
-
+    
+    @objc
+    private func tappedDetailButton() {
+        let actionSheet = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+        
+        let shareAction = UIAlertAction(title: ButtonConstant.shareActionTitle, style: .default) { [weak self] _ in
+            guard let self else { return }
+            self.tappedSharebutton()
+        }
+        
+        let deleteAction = UIAlertAction(title: ButtonConstant.deleteActionTitle, style: .destructive) { [weak self] _ in
+            guard let self else { return }
+            self.createDeleteAlert()
+        }
+        let cancelAction = UIAlertAction(title: ButtonConstant.cancelActionTitle, style: .cancel)
+        
+        [shareAction, deleteAction, cancelAction].forEach(actionSheet.addAction(_:))
```

Comment:
```suggestion
        [shareAction, deleteAction, cancelAction].forEach(actionSheet.addAction)
```

파라미터 없는 함수를 인자로 넘겨줄 때는 이게 깔끔하더라구요 ㅎㅎ